clang/test/Driver/sycl-offload-intelfpga.cpp

```diff
@@ -184,47 +184,39 @@
-// CHK-FPGA-AOCO-PHASES-LIN: 11: compiler, {3}, ir, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 12: backend, {11}, assembler, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 13: assembler, {12}, object, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 14: linker, {13, 9}, ir, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 15: llvm-spirv, {14}, spirv, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 16: backend-compiler, {15}, fpga_aocx, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 17: clang-offload-wrapper, {16}, object, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-LIN: 18: offload, "host-sycl (x86_64-unknown-linux-gnu)" {10}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {17}, image
-
-// CHK-FPGA-AOCO-PHASES-WIN: 9: linker, {8}, image, (host-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 10: compiler, {3}, ir, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 11: backend, {10}, assembler, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 12: assembler, {11}, object, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 13: input, "{{.*}}", archive
-// CHK-FPGA-AOCO-PHASES-WIN: 14: clang-offload-unbundler, {13}, archive
-// CHK-FPGA-AOCO-PHASES-WIN: 15: linker, {12, 14}, ir, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 16: llvm-spirv, {15}, spirv, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 17: backend-compiler, {16}, fpga_aocx, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 18: clang-offload-wrapper, {17}, object, (device-sycl)
-// CHK-FPGA-AOCO-PHASES-WIN: 19: offload, "host-sycl (x86_64-pc-windows-msvc)" {9}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice-coff)" {18}, image
+// CHK-FPGA-AOCO-PHASES: 0: input, "[[INPUTA:.+\.a]]", object, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 1: input, "[[INPUTCPP:.+\.cpp]]", c++, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 3: input, "[[INPUTCPP]]", c++, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 4: preprocessor, {3}, c++-cpp-output, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 5: compiler, {4}, sycl-header, (device-sycl)
+// CHK-FPGA-AOCO-PHASES-LIN: 6: offload, "host-sycl (x86_64-unknown-linux-gnu)" {2}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {5}, c++-cpp-output
+// CHK-FPGA-AOCO-PHASES-WIN: 6: offload, "host-sycl (x86_64-pc-windows-msvc)" {2}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice-coff)" {5}, c++-cpp-output
+// CHK-FPGA-AOCO-PHASES: 7: compiler, {6}, ir, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 8: backend, {7}, assembler, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 9: assembler, {8}, object, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 10: linker, {0, 9}, image, (host-sycl)
+// CHK-FPGA-AOCO-PHASES: 11: compiler, {4}, ir, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 12: backend, {11}, assembler, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 13: assembler, {12}, object, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 14: input, "[[INPUTA]]", archive
+// CHK-FPGA-AOCO-PHASES-LIN: 15: clang-offload-unbundler, {14}, object
+// CHK-FPGA-AOCO-PHASES-WIN: 15: clang-offload-unbundler, {14}, archive
```

Comment:
Just curious, why do the output types differ between the OS'es?

Reply:
We treat the unbundling step of fat static libraries differently on Windows than Linux.  Linux we go through the partial link step while on Windows we do not partial link and just call the unbundler with the archive.

Reply:
Thanks!